felix/nftables/match_builder.go

```diff
@@ -0,0 +1,489 @@
+	return m
+}
+
+func (m nftMatch) NotDestPorts(ports ...uint16) generictables.MatchCriteria {
+	portsString := PortsToMultiport(ports)
+	m.clauses = append(m.clauses, fmt.Sprintf("%s dport != %s", m.protocol(), portsString))
+	return m
+}
+
+func (m nftMatch) SourcePortRanges(ports []*proto.PortRange) generictables.MatchCriteria {
+	portsString := PortRangesToMultiport(ports)
+	m.clauses = append(m.clauses, fmt.Sprintf("%s sport %s", m.protocol(), portsString))
+	return m
+}
+
+func (m nftMatch) NotSourcePortRanges(ports []*proto.PortRange) generictables.MatchCriteria {
+	portsString := PortRangesToMultiport(ports)
+	m.clauses = append(m.clauses, fmt.Sprintf("%s sport != %s", m.protocol(), portsString))
+	return m
+}
+
+func (m nftMatch) DestPortRanges(ports []*proto.PortRange) generictables.MatchCriteria {
+	portsString := PortRangesToMultiport(ports)
+	m.clauses = append(m.clauses, fmt.Sprintf("%s dport %s", m.protocol(), portsString))
+	return m
+}
+
+func (m nftMatch) NotDestPortRanges(ports []*proto.PortRange) generictables.MatchCriteria {
+	portsString := PortRangesToMultiport(ports)
+	m.clauses = append(m.clauses, fmt.Sprintf("%s dport != %s", m.protocol(), portsString))
+	return m
+}
+
+func (m nftMatch) ICMPType(t uint8) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("icmp type %d", t))
+	return m
+}
+
+func (m nftMatch) NotICMPType(t uint8) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("icmp type != %d", t))
+	return m
+}
+
+func (m nftMatch) ICMPTypeAndCode(t, c uint8) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("icmp type %d code %d", t, c))
+	return m
+}
+
+func (m nftMatch) NotICMPTypeAndCode(t, c uint8) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("icmp type != %d code != %d", t, c))
+	return m
+}
+
+func (m nftMatch) ICMPV6Type(t uint8) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("icmpv6 type %d", t))
+	return m
+}
+
+func (m nftMatch) NotICMPV6Type(t uint8) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("icmpv6 type != %d", t))
+	return m
+}
+
+func (m nftMatch) ICMPV6TypeAndCode(t, c uint8) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("icmpv6 type %d code %d", t, c))
+	return m
+}
+
+func (m nftMatch) NotICMPV6TypeAndCode(t, c uint8) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("icmpv6 type != %d code != %d", t, c))
+	return m
+}
+
+// VXLANVNI matches on the VNI contained within the VXLAN header.  It assumes that this is indeed a VXLAN
+// packet; i.e. it should be used with a protocol==UDP and port==VXLAN port match.
+//
+// Note: the -m u32 option is not supported on iptables in NFT mode.
+// https://wiki.nftables.org/wiki-nftables/index.php/Supported_features_compared_to_xtables#u32
+func (m nftMatch) VXLANVNI(vni uint32) generictables.MatchCriteria {
```

Comment:
We dropped this in iptables mode too, I think? Should we just remove the function entirely?